src/cli/bin.ts

```diff
@@ -1,8 +1,71 @@
 import yargs from "yargs";
 import _capitalize from "lodash/capitalize";
+import _cloneDeep from "lodash/cloneDeep";
 import fs from "fs-extra";
 import { migrate_43_to_50 } from "../4.3_to_5.0/migrate_43_to_50";
 import path from "path";
+import { ContentRecord, DataModel } from "@activeviam/activeui-sdk-5.0";
+import { ErrorReport, OutcomeCounters } from "../4.3_to_5.0/migration.types";
+import _fromPairs from "lodash/fromPairs";
+import { emptyUIFolder } from "@activeviam/content-server-initialization-5.0";
+
+type MigrationFunction = (
+  legacyUIFolder: ContentRecord,
+  {
+    migratedUIFolder,
+    counters,
+    errorReport,
+    servers,
+    keysOfWidgetPluginsToRemove,
+    legacyPivotFolder,
+    doesReportIncludeStacks,
+  }: {
+    migratedUIFolder: ContentRecord;
+    counters: OutcomeCounters;
+    errorReport: ErrorReport;
+    servers: {
+      [serverKey: string]: {
+        dataModel: DataModel;
+        url: string;
+      };
+    };
+    keysOfWidgetPluginsToRemove?: string[] | undefined;
+    doesReportIncludeStacks: boolean;
+    legacyPivotFolder?: ContentRecord<any> | undefined;
+  },
+) => Promise<void>;
```

Comment:
Related to my other comment on `migrate_43_to_50`: you intend to do the refactoring that we talked about in another PR right ?
Because for example we shouldn't pass just the UI folder but the full content server.

Reply:
Yes, I'll do it in [this PR](https://github.com/activeviam/activeui-migration/pull/55), if that's fine by you.